onnx/defs/math/defs.cc

```diff
@@ -3392,4 +3392,591 @@ ONNX_OPERATOR_SET_SCHEMA(
             propagateShapeFromInputToOutput(ctx, 0, 1);
           }
         }));
+
+
+template <typename T>
+static T get_scalar_value_from_tensor(const ONNX_NAMESPACE::TensorProto* t) {
+  if (t == nullptr) {
+    return T{};
+  }
+
+  auto data_type = t->data_type();
+  switch (data_type) {
+    case ONNX_NAMESPACE::TensorProto::FLOAT:
+      return static_cast<T>(ONNX_NAMESPACE::ParseData<float>(t).at(0));
+    case ONNX_NAMESPACE::TensorProto::DOUBLE:
+      return static_cast<T>(ONNX_NAMESPACE::ParseData<double>(t).at(0));
+    case ONNX_NAMESPACE::TensorProto::INT32:
+      return static_cast<T>(ONNX_NAMESPACE::ParseData<int32_t>(t).at(0));
+    case ONNX_NAMESPACE::TensorProto::INT64:
+      return static_cast<T>(ONNX_NAMESPACE::ParseData<int64_t>(t).at(0));
+    default:
+      fail_shape_inference("Unsupported input data type of ", data_type);
+  }
+}
+
+static const char* DFT_ver16_doc =
+    R"DOC(Computes the discrete Fourier transform of input.)DOC";
+
+    ONNX_OPERATOR_SET_SCHEMA(DFT,
+        16,
+        OpSchema()
+            .SetDoc(DFT_ver16_doc)
+            .Attr(
+                "onesided",
+                "If onesided is 1, only values for w in [0, 1, 2, ..., floor(n_fft/2) + 1] are returned because "
+                "the real-to-complex Fourier transform satisfies the conjugate symmetry, i.e., X[m, w] = X[m,w]=X[m,n_fft-w]*. "
+                "Note if the input or window tensors are complex, then onesided output is not possible. "
+                "Enabling onesided with real inputs performs a Real-valued fast Fourier transform (RFFT)."
+                "When invoked with real or complex valued input, the default value is 0. "
+                "Values can be 0 or 1.",
+                AttributeProto::INT,
+                static_cast<int64_t>(0))
+            .Attr(
+                "axis",
+                "The axis on which to perform the DFT. By default this value is set to 0, which corresponds to the first dimension after the batch index."
+                "This value must be less than signal_dimN, where signal_dimN is the number of dimensions in the signal.",
```

Comment:
Ideally, it should allow negative values, just like other ops.

Reply:
Updated to allow negative values. The axis is defaulted to 1.